Only compute new_untracked_symbols and new_unbacked_bindings if needed.

laithsakka opened the pull request:
Summary:
237s -> 198.. 
buck2 run fbcode//mode/opt fbcode//torchrec/distributed/tests:pt2_compile_benchmark -- --num-features=2000

Test Plan: NA

Differential Revision: D65638637




cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv @voznesenskym @penguinwu @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @jiayisunx @chenyang78 @kadeng @chauhang @amjames

isuruf commented on torch/fx/passes/runtime_assert.py:
```diff
@@ -350,27 +350,29 @@ def match_symbol(symint, cb):
             ):
                 # this guards against deleting calls like item() that produce new untracked symbols
-                new_untracked_symbols = sym_expr.free_symbols - expr_to_proxy.keys()
+                def new_untracked_symbols():
+                    return sym_expr.free_symbols - expr_to_proxy.keys()
```
Maybe do
```suggestion
                def has_new_untracked_symbols():
                    return bool(sym_expr.free_symbols - expr_to_proxy.keys())
```
to make it clear what it's used for

laithsakka commented:
mad it even better.

isuruf commented on torch/fx/passes/runtime_assert.py:
```diff
@@ -350,27 +350,36 @@ def match_symbol(symint, cb):
-                new_untracked_symbols = sym_expr.free_symbols - expr_to_proxy.keys()
+                def has_new_untracked_symbols():
+                    for symbol in sym_expr.free_symbols :
+                        if symbol not in expr_to_proxy.keys():
```
```suggestion
                        if symbol not in expr_to_proxy:
```

laithsakka committed:
only compute new_untracked_symbols() and new_unbacked_bindings() if needed. (#140083)

Summary:

237s -> 198.. 
buck2 run fbcode//mode/opt fbcode//torchrec/distributed/tests:pt2_compile_benchmark -- --num-features=2000

Test Plan: NA

Reviewed By: ezyang

Differential Revision: D65638637

laithsakka commented:
@pytorchbot merge
